src/Npgsql/TypeHandlers/TextHandler.cs

```diff
@@ -169,6 +169,9 @@ ValueTask<ArraySegment<char>> INpgsqlTypeHandler<ArraySegment<char>>.Read(Npgsql
         }
 
         ValueTask<byte[]> INpgsqlTypeHandler<byte[]>.Read(NpgsqlReadBuffer buf, int byteLen, bool async, FieldDescription fieldDescription)
+            => ReadBytes(buf, byteLen, async);
+
+        internal virtual ValueTask<byte[]> ReadBytes(NpgsqlReadBuffer buf, int byteLen, bool async)
```

Comment:
Same question as for `ValidateAndGetLengthBytes` - why not just have the code in `INpgsqlTypeHandler<byte[]>.Read()`?